### PR TITLE
nixos/tests: add a test for config.users.mutableUsers.

### PR DESCRIPTION
**Pull request by gleber**
It's in preparation to add an assertion for #4990.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by grahamc** [comment]
Hey @gleber you might check out my work here: https://github.com/NixOS/nixpkgs/pull/31034

**Comment by gleber** [comment]
As mentioned in #31034, I think this PR is useful on its own. Waiting for review

File: nixos/tests/mutable-users.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  testScript = {nodes, ...}: let
+    immutableSystem = nodes.machine.config.system.build.toplevel;
+    mutableSystem = nodes.mutable.config.system.build.toplevel;
+  in ''
+    $machine->start();
+    $machine->waitForUnit("default.target");
+
+    # Machine starts in immutable mode. Add a user and test if reactivating
+    # configuration removes the user.
+    $machine->fail("cat /etc/passwd | grep foobar");
```
Comment (obadz) [inline nixos/tests/mutable-users.nix]: Might want to make that `grep ^foobar:` instead

**Comment by obadz** [comment]
Added a comment on the code. Otherwise, LGTM. Given that this isn't part of `release.nix`, I don't think the hurdle to clear should be too high… I'm happy to merge once comment addressed.

**Comment by obadz** [comment]
LGTM. Deferring to @fpletz.

**Comment by gleber** [comment]
@GrahamcOfBorg @grahamc  I've ran `nix-env --file ~/code/nixpkgs --query --available --json` and it succeeds. I'm not sure how else to debug that failed check...

**Comment by fpletz** [comment]
Thanks!

**Comment by grahamc** [comment]
When seeing why the tests fail, make sure to be rebased on a recent master. That is how the tests are run, and sometimes this is caused by breakage already on master.

**Pull request merged**